keel-sqs/src/main/kotlin/com/netflix/spinnaker/keel/sqs/SqsResourceCheckListener.kt

```diff
@@ -27,38 +34,48 @@ internal class SqsResourceCheckListener(
   private val objectMapper: ObjectMapper,
   private val actuator: ResourceActuator
 ) {
-  private lateinit var listenerJob: Job
+  private lateinit var rootJob: Job
 
-  private val scope: CoroutineScope
-    get() = CoroutineScope(Default)
+  private val scope: CoroutineScope = CoroutineScope(IO)
 
   @PostConstruct
   fun startListening() {
-    listenerJob = scope.launch {
-      while (isActive) {
-        handleMessages()
+    rootJob = scope.launch {
+      val channel = Channel<Message>()
+      launchMessageReceiver(channel)
+      repeat(10) {
```

Comment:
maybe throw this number in a `@Value` or `@ConfigurationProperties`?

Reply:
Yeah. I meant to. Then I forgotted.